File: RandomNumberCountJava/src/main/java/org/eduami/hadoop/dataset/randomnumbercountjava/RandomNumberCount.java
package org.eduami.hadoop.dataset.randomnumbercountjava;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

/**
 * Counts number of random numbers in file Reads dataset file
 *
 */
public class RandomNumberCount {
	public static void main(String[] args) throws FileNotFoundException {
		long startTime = System.currentTimeMillis();
		Scanner scanner = new Scanner(new File("dataset-small.txt"));
		Map<Integer, Integer> countMap = new HashMap<Integer, Integer>();
		while (scanner.hasNextInt()) {
			int number = scanner.nextInt();
			if (countMap.containsKey(number)) {
				int value = countMap.get(number);
				countMap.put(number, ++value); // increment occurrence
			} else {
				countMap.put(number, 1);// Initialize
			}
		}
		System.out.println(countMap);
		scanner.close();
		long endTime = System.currentTimeMillis();
		System.out.println("Time taken in milli seconds: " + (endTime - startTime));
	}
}

File: RandomNumberCountPython/RandomNumberCount.py
'''
Created on 11-Mar-2017

@author: mario
'''
import collections
import time


if __name__ == '__main__':
        start_time = time.time()
        countDict = {}
        with open('dataset-small.txt') as f:  # Open file
            for line in f:  # read number
                number = int(line)
                if countDict.has_key(number):  # if number found increment it
                    countDict[number] = countDict[number] + 1 
                else:
                    countDict[number] = 1  # initialize
        print countDict
        print("Time taken in milli seconds: %s " % (time.time() - start_time)) 
        # print collections.OrderedDict(sorted(countDict.items())) # sort by key and print
